x-pack/legacy/plugins/ml/public/jobs/new_job_new/pages/index_or_search/page.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC } from 'react';
+import {
+  EuiPage,
+  EuiPageBody,
+  EuiTitle,
+  EuiPageHeader,
+  EuiPageHeaderSection,
+  EuiPageContent,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { SavedObjectFinder } from 'ui/saved_objects/components/saved_object_finder';
+
+export interface PageProps {
+  nextStepPath: string;
+}
+
+export const Page: FC<PageProps> = ({ nextStepPath }) => {
+  const RESULTS_PER_PAGE = 20;
+
+  return (
+    <EuiPage data-test-subj="mlPageSourceSelection">
+      <EuiPageBody restrictWidth={1200}>
+        <EuiPageHeader>
+          <EuiPageHeaderSection>
+            <EuiTitle size="m">
+              <h1>
+                <FormattedMessage
+                  id="xpack.ml.newJob.wizard.createFromNewSearchTitle"
+                  defaultMessage="From a New Search, Select Index"
+                />{' '}
+                <FormattedMessage
+                  id="xpack.ml.newJob.wizard.createFromSavedSearchTitle"
+                  defaultMessage="Or, From a Saved Search"
+                />
+              </h1>
+            </EuiTitle>
+          </EuiPageHeaderSection>
+        </EuiPageHeader>
+        <EuiPageContent>
+          <SavedObjectFinder
```

Comment:
would it be possible to reuse the `SearchSelection` component from visualize?

Reply:
`SearchSelection` intended to be used with modal window, also there is no way to modify the header